mark Control Center backups as a commercial feature (#4396)
vaadin/docs: articles/control-center/database/backups.adoc

Enable backups by setting the `postgres.backups.enabled` value to `true` when installing Control Center.
Enabling backups does nothing by itself.
You must also select a backup method and provide proper configuration for the selected method, as well as creating a Kubernetes secret with a valid Vaadin commercial license key.
The available methods for backups are:

- <<Volume Snapshots>>
- <<Barman Object Stores>>

To decide which method to use, it is recommended to read https://cloudnative-pg.io/documentation/1.25/backup/#object-stores-or-volume-snapshots-which-one-to-use[CloudNativePG's official documentation].

=== Create License Secret

> Since backups are a commercial feature, you must provide a valid Vaadin pro license key to enable the feature.

Reviewer: There's a couple of references to "pro" key in this section, but this is not synonymous with commercial license. Are only Pro keys allowed? If not, then we should make this wording generic.

Let's also mention that if they are using an offline key, it might need to be downloaded from their Vaadin account. We could link to the licenses page (https://vaadin.com/docs/latest/flow/configuration/licenses), where this is explained.

- `ACCESS_KEY_ID`: the ID of the access key that will be used to upload files into S3
- `ACCESS_SECRET_KEY`: the secret part of the access key mentioned above

> WARNING: To prevent the sensitive values of the secrets to be *stored in the shell's history*, write down the values in files *without an extra newline character* at the end of the text and pass those files to `kubectl` to save their content as the secret's values.

Reviewer: ```suggestion
WARNING: To prevent the sensitive values of the secrets being *stored in the terminal command history*, write down the values in files (*without* an extra newline character at the end of the text) and pass those files to `kubectl` to save their content as the secret's values.
```

When using either *Storage account access key* or *Storage account SAS Token*, the credentials need to be stored inside a Kubernetes Secret, adding data entries only when needed.
The following command performs that:

> WARNING: To prevent the sensitive values of the secrets to be *stored in the shell's history*, write down the values in files *without an extra newline character* at the end of the text and pass those files to `kubectl` to save their content as the secret's values.

Reviewer: ```suggestion
WARNING: To prevent the sensitive values of the secrets being *stored in the terminal command history*, write down the values in files (*without* an extra newline character at the end of the text) and pass those files to `kubectl` to save their content as the secret's values.
```
